src/codeu/chat/util/Serializers.java

```diff
@@ -151,6 +154,68 @@ public void write(OutputStream out, Collection<T> value) throws IOException {
     };
   }
 
+  public static <K, V> Serializer<Map<K, V>> map(final Serializer<K> keySerializer, final Serializer<V> valueSerializer) {
+    return new Serializer<Map<K, V>>() {
+
+      @Override
+      public void write(OutputStream out, Map<K, V> value) throws IOException {
+        INTEGER.write(out, value.size());
+        Collection<K> keys = new ArrayList<>(value.size());
+        Collection<V> values = new ArrayList<>(value.size());
+        for (final K x : value.keySet()) {
+          V v = value.get(x);
+          keys.add(x);
+          values.add(v);
+        }
+        Serializers.collection(keySerializer).write(out, keys);
+        Serializers.collection(valueSerializer).write(out, values);
+      }
+
+      @Override
+      public Map<K, V> read(InputStream in) throws IOException {
+        final int size = INTEGER.read(in);
+        List<K> keys = new ArrayList<>(Serializers.collection(keySerializer).read(in));
+        List<V> values = new ArrayList<>(Serializers.collection(valueSerializer).read(in));
+        Map<K, V> hashMap = new HashMap<>(size);
+        for (int i = 0; i < size; i++) {
+          hashMap.put(keys.get(i), values.get(i));
+        }
+        return hashMap;
+      }
+    };
+  }
+  
+  public static <K, V> Serializer<HashMap<K, V>> HashMap(final Serializer<K> keySerializer, final Serializer<V> valueSerializer) {
```

Comment:
Can we use the map serializer instead? If we really can't, this should also be in CAPS.